### PR TITLE
Feature: S3 support

### PR DESCRIPTION
**Pull request by phil1995**
Adds support for S3 as a cloud provider.

Since S3 does not follow the traditional user:pass (+ possibly URL) scheme, a display name has been added to easily distinguish S3 credentials.

### CODE REVIEW

File: Cryptomator/S3/S3LoginView.swift [hunk 1]
```diff
@@ -0,0 +1,78 @@
...
+import CryptomatorCloudAccessCore
+import CryptomatorCommonCore
+import SwiftUI
+
+struct S3LoginView: View {
+	@FocusStateLegacy private var focusedField: S3LoginField? = .displayName
+	@ObservedObject var viewModel: S3LoginViewModel
+	var body: some View {
+		Form {
+			TextField("Display Name",
```
Comment (tobihagemann) [inline Cryptomator/S3/S3LoginView.swift:19]: Requires localization.

File: Cryptomator/S3/S3LoginViewController.swift [hunk 2]
```diff
@@ -0,0 +1,67 @@
...
+		let cancelButton = UIBarButtonItem(barButtonSystemItem: .cancel, target: self, action: #selector(cancel))
+		navigationItem.leftBarButtonItem = cancelButton
+		title = viewModel.title
+
+		viewModel.saveDisabled.sink { disabled in
+			doneButton.isEnabled = !disabled
+		}.store(in: &subscriptions)
+
+		let hud = ProgressHUD()
+		hud.text = LocalizedString.getValue("webDAVAuthentication.progress")
```
Comment (tobihagemann) [inline Cryptomator/S3/S3LoginViewController.swift:42]: Value might be correct but we should use a more specific key. 😁

**Review by tobihagemann — CHANGES_REQUESTED** [review]

File: Cryptomator/S3/S3LoginView.swift (hunk 1 above)
Comment (phil1995) [reply Cryptomator/S3/S3LoginView.swift]: Resolved by 172358b852591303ee737af593f8dd403c4fcc0b.

File: Cryptomator/S3/S3LoginViewController.swift (hunk 2 above)
Comment (phil1995) [reply Cryptomator/S3/S3LoginViewController.swift]: Resolved by bc378db4408b121e7a9405af30e13078012dbc84.

File: Cryptomator/S3/S3LoginViewModel.swift [hunk 3]
```diff
@@ -0,0 +1,122 @@
...
+	case error(Error)
+	case verifyingCredentials
+	case loggedIn(S3Credential)
+	case notLoggedIn
+}
+
+enum S3LoginViewModelError: Error {
+	case emptyField
+	case invalidEndpoint
+	case displayNameAlreadyExists(String)
```
[the remark this answers is not in the document]
Comment (phil1995) [reply Cryptomator/S3/S3LoginViewModel.swift]: Resolved by d05c321c920dcdd121f10326352b624166b5f4bd and 37d6ca090b69dce015e6d210e4fcf5b06af5ad34.

File: CryptomatorCommon/Sources/CryptomatorCommonCore/Manager/CloudProviderType.swift [hunk 4]
```diff
@@ -16,6 +16,7 @@ public enum CloudProviderType: Codable, Equatable, Hashable {
 	case pCloud
 	case webDAV(type: WebDAVType)
 	case localFileSystem(type: LocalFileSystemType)
+	case s3
```
[the remark this answers is not in the document]
Comment (phil1995) [reply CryptomatorCommon/Sources/CryptomatorCommonCore/Manager/CloudProviderType.swift]: Resolved by 1156b8fbf8db31fc973a13d2a8ae3abe3a6deeab.

**Pull request merged**